nmap.go

```diff
@@ -94,6 +98,55 @@ func (s *Scanner) Streamer(stream io.Writer) *Scanner {
 	return s
 }
 
+// Version returns the version of the installed nmap binary
+func (s *Scanner) Version() (result any, err error) {
+	s.args = append(s.args, "-V")
+	args := s.args
+	cmd := exec.CommandContext(s.ctx, s.binaryPath, args...)
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(stdout)
+	for scanner.Scan() {
+		m := scanner.Text()
+		splitted := strings.Split(m, " ")
+		if len(splitted) >= 2 {
+			result = splitted[2]
+			break
+		}
+	}
+
+	errScanner := bufio.NewScanner(stderr)
+	var errorOutput = []string{}
+	for errScanner.Scan() {
+		m := errScanner.Text()
+		errorOutput = append(errorOutput, m)
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		return
+	}
+
+	if len(errorOutput) > 0 {
+		return nil, fmt.Errorf("%v", errorOutput)
+	}
```

Comment:
What is the purpose of this code? To check that we can run the binary without errors on top of getting the version info? If yes, these should be two separate functions IMO.

Reply:
I will check all your comments and will make the proposed changes in the first calendar week of december. 
I'm really busy at the moment.

Reply:
No rush, take all the time you need! :) Thanks again!